regression_pred_result.py

```diff
@@ -0,0 +1,84 @@
+import numpy as np
+import torch
+
+
+class RegressionPredictResult:
+    def __init__(self, res: {}):
+        # res:
+        # {
+        #   criterion <class 'tabpfn.model.bar_distribution.FullSupportBarDistribution'>
+        #   mean <class 'numpy.ndarray'>
+        #   median <class 'numpy.ndarray'>
+        #   mode <class 'numpy.ndarray'>
+        #   logits <class 'numpy.ndarray'>
+        #   buckets <class 'numpy.ndarray'>
+        #   quantile_0.10 <class 'numpy.ndarray'>
+        #   quantile_0.20 <class 'numpy.ndarray'>
+        #   quantile_0.30 <class 'numpy.ndarray'>
+        #   quantile_0.40 <class 'numpy.ndarray'>
+        #   quantile_0.50 <class 'numpy.ndarray'>
+        #   quantile_0.60 <class 'numpy.ndarray'>
+        #   quantile_0.70 <class 'numpy.ndarray'>
+        #   quantile_0.80 <class 'numpy.ndarray'>
+        #   quantile_0.90 <class 'numpy.ndarray'>
+        # }
+
+        self.mean = res["mean"]
+        self.median = res["median"]
+        self.mode = res["mode"]
+        self.quantiles = {k: v for k, v in res.items() if k.startswith("quantile_")}
+
+        # assume values are either all numpy arrays or all torch tensors
+        if isinstance(self.mean, torch.Tensor):
+            self._val_type = torch.Tensor
+        elif isinstance(self.mean, np.ndarray):
+            self._val_type = np.ndarray
+        elif isinstance(self.mean, list):
+            self._val_type = list
+        else:
+            raise ValueError(f"Invalid type for mean: {type(self.mean)}")
+
+        # assert all values are of the same type
+        for val in [self.mean, self.median, self.mode, *self.quantiles.values()]:
+            assert isinstance(val, self._val_type)
+
+    @property
+    def val_type(self):
+        return self._val_type
+
+    @staticmethod
+    def serialize(res: "RegressionPredictResult") -> {str: list}:
```

Comment:
i think serialize is a slightly bad name here, as that usually implies to str. Maybe something like to `to_basic_representation` and `from_basic_representation`?

Reply:
Yup, I used `to_basic_representation` and `from_basic_representation` as per suggested :)